pkgs/top-level/config.nix

```diff
@@ -0,0 +1,204 @@
+
+      headless = mkUse {
+        default = false;
+        description = "Whether to build nixpkgs packages for a headless system.";
+      };
+
+      /* Graphical toolkits */
+
+      noGraphics = mkUse {
+        default = use.headless;
+        description = "Whether to build nixpkgs without support for any graphical toolkits.";
+      };
+
+      X = mkUse {
+        feature = "X11";
+        default = !use.noGraphics;
+        extra = whenDisabled {
+          x11Support = false;
+          withX = false;
+        };
+      };
+
+      wayland = mkUse {
+        feature = "Wayland";
+        default = !use.noGraphics;
+        extra = whenDisabled {
+          waylandSupport = false;
+          withWayland = false;
+        };
+      };
+
+      gtk = mkUse {
+        feature = "GTK";
+        default = use.X || use.wayland;
+        extra = whenDisabled {
+          withGtk = false;
+          gtkSupport = false;
+          withGtk2 = false;
+          gtk2Support = false;
+          withGtk3 = false;
+          gtk3Support = false;
+          gtk2 = null;
+          gtk3 = null;
+        };
+      };
+
+      qt = mkUse {
+        feature = "Qt";
+        default = use.X || use.wayland;
+        extra = whenDisabled {
+          withQt = false;
+          qtSupport = false;
+          withQt3 = false;
+          qt3Support = false;
+          withQt4 = false;
+          qt4Support = false;
+          withQt5 = false;
+          qt5Support = false;
+          qt3 = null;
+          qt4 = null;
+          qt5 = null;
+        };
+      };
+
+      /* Audio stuff */
+
+      noAudio = mkUse {
+        default = use.headless;
+        description = "Whether to build nixpkgs without support for any audio toolkits.";
+      };
+
+      alsa = mkUse {
+        feature = "ALSA";
+        default = !use.noAudio && localSystem.isLinux;
+        defaultText = "!noAudio && isLinux";
+        extra = whenDisabled {
+          alsaSupport = false;
+          withAlsa = false;
+          alsaLib = null;
+        };
+      };
+
+      pulseaudio = mkUse {
+        feature = "PulseAudio";
+        default = !use.noAudio;
+        defaultText = "!noAudio";
+        extra = whenDisabled {
+          pulseSupport = false;
+          withPulse = false;
+          pulseaudioSupport = false;
+          withPulseAudio = false;
+          libpulseaudio = null;
+          pulseaudio = null;
+          pulseaudioFull = null;
+        };
+      };
+
+      /* Desktop environment integration */
+
+      gnome = mkUse {
+        feature = "GNOME integration";
+        default = use.gtk;
+        extra = whenDisabled {
+          gnomeSupport = false;
+          withGnome = false;
+        };
+      };
+
+      kde = mkUse {
+        feature = "KDE integration";
+        default = use.qt;
+        extra = whenDisabled {
+          withKDE = false;
+        };
+      };
+
+    };
+
+  };
+
+  imports = [
+    (mkAliasOptionModule [ "use" "x11" ] [ "use" "X" ])
```

Comment:
I don't think we need to introduce preliminary aliases. We can always add them later if they're really wanted.

Reply:
and I like that!